cmd/elasticindexer/main.go

```diff
@@ -104,6 +108,19 @@ func startIndexer(ctx *cli.Context) error {
 	return nil
 }
 
+func requestSettings(host wsindexer.WSClient, retryDurationInSec uint32) {
+	for {
+		err := host.Send([]byte{}, outport.TopicSettings)
+		if err != nil {
+			time.Sleep(time.Duration(retryDurationInSec) * time.Second)
```

Comment:
please refactor this to allow the normal closing of the node if the settings were not received in time.